File: archive/number-of-strings-that-appear-as-substrings-in-word.js
const numOfStrings = function(patterns, word) {
  let ans = 0
  patterns.forEach(el => {
    if (word.length !== word.split(el).join('').length) {
      ans++
    }
  })

  return ans
}

console.log(numOfStrings(["a","abc","bc","d"], "abc"))

File: number-of-strings-that-appear-as-substrings-in-word.go
func numOfStrings(patterns []string, word string) int {
    ans:=0

    for i:=0; i < len(patterns); i++ {
        spl:= strings.Join(strings.Split(word, patterns[i]), "")

        if (len(spl) != len(word)) {ans++}
    }

    return ans
}